res/controllers/Pioneer-DDJ-400-script.js

```diff
@@ -0,0 +1,774 @@
+// Pioneer-DDJ-400-script.js
+// ****************************************************************************
+// * Mixxx mapping script file for the Pioneer DDJ-400.
+// * Author: Warker, nschloe, dj3730
+// * Forum: https://mixxx.org/forums/viewtopic.php?f=7&t=12113
+// * Wiki: https://www.mixxx.org/wiki/doku.php/pioneer_ddj-400
+//
+// Upstream MIDI spec:
+// * https://www.pioneerdj.com/-/media/pioneerdj/software-info/controller/ddj-400/ddj-400_midi_message_list_e1.pdf
+//
+//             Working:
+//                 * Mixer Section (Faders, EQ, Filter, Gain, Cue)
+//                 * Browsing and loading + Waveform zoom (shift)
+//                 * Jogwheels, Scratching, Bending
+//                 * cycle Temporange
+//                 * Beat Sync
+//                 * Beat Loop Mode
+//                 * Sampler Mode
+//                 * BeatFX (controls Effect Unit 1.  LEFT selects EFFECT1, RIGHT selects EFFECT2, FX_SELECT selects EFFECT3.
+//                   ON/OFF toggles selected effect slot.  SHIFT+ON/OFF disables all three effect slots.
+//                 * Hot Cue Mode
+//
+//             Partially:
```

Comment:
Are these all still problems?

Reply:
The Beatjump and PadFX issues still exist.  The Output (lights) stuff is mainly functional, so I think it would be fair to remove that line.  The Loop Section issues still exist, most notably the '4 beat loop' issue mentioned in your other comment relating to "loopin4beatPressedLong".

Reply:
What is not working about beatjump? Can you fix it?

Reply:
Beatjump had outstanding functionality which is now added